File: android/app/src/main/java/com/thinsia/ubivault/util/BindingAdapters.java
package com.thinsia.ubivault.util;

import android.view.View;
import androidx.databinding.BindingAdapter;

public class BindingAdapters {

    @BindingAdapter("show")
    public static void visibleWhenNotNull(View view, Object value) {
        boolean hidden = value == null;
        if (value instanceof Boolean) {
            hidden = !(boolean) value;
        }
        view.setVisibility(hidden ? View.GONE : View.VISIBLE);
    }

    @BindingAdapter("hide")
    public static void hideWhenNotNull(View view, Object value) {
        boolean visible = value == null;
        if (value instanceof Boolean) {
            visible = !(boolean) value;
        }
        view.setVisibility(visible ? View.VISIBLE : View.GONE);
    }

}

File: android/app/src/main/java/com/thinsia/ubivault/util/BindingAdapters.kt
package com.thinsia.ubivault.util

import android.view.View
import androidx.databinding.BindingAdapter

@BindingAdapter("show")
fun View.visibleWhenNotNull(value: Any?) {
    var hidden = value == null
    if (value is Boolean) {
        hidden = !value
    }
    visibility = if (hidden) View.GONE else View.VISIBLE
}

@BindingAdapter("hide")
fun View.hideWhenNotNull(value: Any?) {
    var visible = value == null
    if (value is Boolean) {
        visible = !value
    }
    visibility = if (visible) View.VISIBLE else View.GONE
}
